Assets/LoggingManager/LoggingManager.cs

```diff
@@ -126,220 +95,147 @@ public void SaveLog(string collectionLabel)
-                {
-                    if (line != "")
-                    {
-                        line += fieldSeperator;
-                    }
-
-                    if (log.Value.TryGetValue(i, out temp))
-                    {
-                        line += ConvertToString(temp);
-                    }
-                    else
-                    {
-                        line += "NULL";
-                    }
-                }
-                file.WriteLine(line);
-            }
+            Debug.LogError("Collection " + collectionLabel + " does not exist.");
         }
-        Debug.Log(label + " logs with " + collections[label].count + 1 + " rows saved to " + savePath);
     }
 
 
-    // Generates the headers in a CSV format and saves them to the CSV file
-    private string GenerateHeaders(LogCollection collection)
+    public void DeleteAllLogs()
     {
-        string headers = "";
-        foreach (string key in collection.log.Keys)
+        foreach (var keyValuePair in logsList)
         {
-            if (headers != "")
-            {
-                headers += fieldSeperator;
-            }
-            headers += key;
+            logsList.Remove(keyValuePair.Key);
         }
-        return headers;
     }
 
-    private void SaveToSQL(string label)
+    // Formats the logs to a CSV row format and saves them. Calls the CSV headers generation beforehand.
+    // If a parameter doesn't have a value for a given row, uses the given value given previously (see 
+    // UpdateHeadersAndDefaults).
+    private void SaveToCSV(string label)
     {
-        if (!enableMySQLSave) { return; }
-
-        if (!collections.ContainsKey(label))
+        if (!enableCSVSave) return;
+        if (logsList.TryGetValue(label, out LogStore logStore))
         {
-            Debug.LogError("Could not find collection " + label + ". Aborting.");
-            return;
+            WriteToCSV writeToCsv = new WriteToCSV(logStore, savePath, filePrefix, fileExtension);
+            writeToCsv.WriteAll();
         }
+    }
 
-        if (collections[label].log.Keys.Count == 0)
-        {
-            Debug.LogError("Collection " + label + " is empty. Aborting.");
-            return;
-        }
+    private void SaveToSQL(string label)
+    {
+        //if (!enableMySQLSave) { return; }
 
-        connectToMySQL.AddToUploadQueue(collections[label].log, collections[label].label);
-        connectToMySQL.UploadNow();
+        //if (!logsList.ContainsKey(label)) {
+        //    Debug.LogError("Could not find collection " + label + ". Aborting.");
+        //    return;
+        //}
+
+        //if (logsList[label].log.Keys.Count == 0) {
+        //    Debug.LogError("Collection " + label + " is empty. Aborting.");
+        //    return;
+        //}
+
+        //connectToMySQL.AddToUploadQueue(logsList[label].log, logsList[label].label);    
+        //connectToMySQL.UploadNow();
```

Comment:
Does this mean that the current commit completely removes saving to SQL?
I guess you are doing this due to incompatibility issues?

Reply:
Yes, when we created this pull request it wasn't planned to update the MySQL part.
The problem is that the former way of logging data to SQL used a `Dictionary<string, Dictionary<int, object>>`  while now we log everything using strings with the `ExportAll<string>()` [function](https://github.com/Vincent-SD/ArduinoLogger/blob/42af7ff23d76278f7d092e34623cafa8c4abccc3/Assets/LoggingManager/LogStore.cs#L187).

If you want me to update the MySQL part as well, would you prefer that I make the changes in the "Adapting LoggingManager" commit or in a new one ?